lib/ansible/modules/windows/win_user_right.ps1

```diff
@@ -0,0 +1,354 @@
+        if ($domain -eq $env:COMPUTERNAME) {
+            $adsi = [ADSI]("WinNT://$env:COMPUTERNAME,computer")
+            $group = $adsi.psbase.children | Where-Object { $_.schemaClassName -eq "group" } | Where-Object { $_.Name -eq $username }
+        } else {
+            $group = $null
+        }
+        if ($group) {
+            $account = New-Object System.Security.Principal.NTAccount($username)
+        } else {
+            $account = New-Object System.Security.Principal.NTAccount($domain, $username)
+        }
+    } else {
+        # when in a domain NTAccount(String) will favour domain lookups check
+        # if username is a local user and explictly search on the localhost for
+        # that account
+        $adsi = [ADSI]("WinNT://$env:COMPUTERNAME,computer")
+        $user = $adsi.psbase.children | Where-Object { $_.schemaClassName -eq "user" } | Where-Object { $_.Name -eq $username }
+        if ($user) {
+            $account = New-Object System.Security.Principal.NTAccount($env:COMPUTERNAME, $username)
+        } else {
+            $account = New-Object System.Security.Principal.NTAccount($username)
+        }
+    }
+    
+    try {
+        $account_sid = $account.Translate([System.Security.Principal.SecurityIdentifier])
+    } catch {
+        Fail-Json $result "Account Name: $account_name is not a valid account, cannot get SID: $($_.Exception.Message)"
+    }
+    
+    $account_sid.Value
+}
+
+Function Run-SecEdit($arguments) {
+    $rc = $null
+    $stdout = $null
+    $stderr = $null
+    $log_path = [IO.Path]::GetTempFileName()
+    $arguments = $arguments + @("/log", $log_path, "/areas", "USER_RIGHTS")
+
+    try {
+        $stdout = &SecEdit.exe $arguments | Out-String
+    } catch {
+        $stderr = $_.Exception.Message
+    }
+    $log = Get-Content -Path $log_path
+    Remove-Item -Path $log_path -Force
+
+    $return = @{
+        log = ($log -join "`n").Trim()
+        stdout = $stdout
+        stderr = $stderr
+        rc = $LASTEXITCODE
+    }
+
+    $return
+}
+
+Function Compare-List($existing_list, $new_list) {
+    $comparison = @{
+        mismatch = $false
+        added = @()
+        removed = @()
+    }
+
+    foreach ($entry in $existing_list) {
+        if ($new_list -notcontains $entry) {
+            $comparison.removed += Get-Username -sid $entry
+            $comparison.mismatch = $true
+        }
+    }
+    foreach ($entry in $new_list) {
+        if ($existing_list -notcontains $entry) {
+            $comparison.added += Get-Username -sid $entry
+            $comparison.mismatch = $true
+        }
+    }
+
+    $comparison
+}
+
+Function Build-NewUserList($action, $users, $existing_users) {
+    $new_users = @()
+    if ($action -eq "remove") {
+        foreach ($existing_user in $existing_users) {
+            $user_match = $true
+            foreach ($user in $users) {
+                $user_sid = Get-SID -account_name $user
```

Comment:
This is potentially expensive if either list is large, since you have to re-lookup the specified user SIDs on each value . Might be better to do a single-pass over each upfront to build a HashSet and use set theory methods to find users to remove.